test/test_backend/db/test_utils.py

```diff
@@ -25,17 +25,27 @@ class FetchAuditTestRunner(unittest.TestCase):
     mock_data: List[Dict] = None
     con: Connection = None
 
+    @staticmethod
+    def clean_fetch_audit(con: Connection):
+        """Remove any rows that should not be there.
+
+        Normally this would only need to be done at the end in tearDown(), but in cases of errors, rows could sneak in
+        and not be removed at the end, so we should also clean in setUp()."""
+        run_sql(con, f"DELETE FROM fetch_audit WHERE comment LIKE 'Unit testing.%';")
+
+
     @classmethod
     def setUpClass(cls):
         """setUp"""
         cls.con = get_db_connection(schema='')
+        cls.clean_fetch_audit(cls.con)
```

Comment:
## Bug fix: Edge cases where errors cause permanent state change, causing tests to fail
During some tests, temporary rows get inserted into `fetch_audit`. They're removed during `tearDown()`. However, in cases where there were errors during a run, `tearDown()` doesn't get to run, leaving an unclean state in which the tests would continue to fail indefinitely. I've added this cleanup step to the `setUp()` now.